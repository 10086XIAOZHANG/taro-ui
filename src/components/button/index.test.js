import Nerv, { findDOMNode } from 'nervjs'
import { renderToString } from 'nerv-server'
import { Simulate, renderIntoDocument } from 'nerv-test-utils'

import AtButton from '../../../.temp/components/button/index'

describe('AtButton Snap', () => {
  it('render AtButton -- props size(normal)', () => {
    const componet = renderToString(<AtButton size='normal'>按钮</AtButton>)
    expect(componet).toMatchSnapshot()
  })

  it('render AtButton -- props size(small)', () => {
    const componet = renderToString(<AtButton size='small'>按钮</AtButton>)
    expect(componet).toMatchSnapshot()
  })

  it('render AtButton -- props type(primary)', () => {
    const componet = renderToString(<AtButton type='primary'>按钮</AtButton>)
    expect(componet).toMatchSnapshot()
  })

  it('render AtButton -- props type(secondary)', () => {
    const componet = renderToString(<AtButton type='secondary'>按钮</AtButton>)
    expect(componet).toMatchSnapshot()
  })

  it('render AtButton -- props circle', () => {
    const componet = renderToString(<AtButton circle>按钮</AtButton>)
    expect(componet).toMatchSnapshot()
  })

  it('render AtButton -- props loading', () => {
    const componet = renderToString(<AtButton loading>按钮</AtButton>)
    expect(componet).toMatchSnapshot()
  })

  it('render AtButton -- props disabled', () => {
    const componet = renderToString(<AtButton disabled>按钮</AtButton>)
    expect(componet).toMatchSnapshot()
  })
})

describe('AtButton Event', () => {
  it('AtButton onClick', () => {
    const onClick = jest.fn()

    const component = renderIntoDocument(<AtButton onClick={onClick}>按钮</AtButton>)
    const componentDom = findDOMNode(component, 'at-button')

    Simulate.click(componentDom)
    expect(onClick).toBeCalled()
  })

  it('AtButton disabled, onClick not to be called', () => {
    const onClick = jest.fn()

    const component = renderIntoDocument(<AtButton disabled onClick={onClick}>按钮</AtButton>)
    const componentDom = findDOMNode(component, 'at-button')

    Simulate.click(componentDom)
    expect(onClick).not.toBeCalled()
  })
})
