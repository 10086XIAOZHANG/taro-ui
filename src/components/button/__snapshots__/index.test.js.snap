// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtButton Snap render AtButton -- props circle 1`] = `"<div class=\\"at-button at-button--normal at-button--circle\\">undefined<div class=\\"at-button__text\\">按钮</div></div>"`;

exports[`AtButton Snap render AtButton -- props disabled 1`] = `"<div class=\\"at-button at-button--normal at-button--disabled\\">undefined<div class=\\"at-button__text\\">按钮</div></div>"`;

exports[`AtButton Snap render AtButton -- props loading 1`] = `"<div class=\\"at-button at-button--normal at-button--icon\\"><div class=\\"at-button__icon\\"><span style=\\"font-size:20px;color:;\\" class=\\"taro-text at-icon at-icon-loading\\"></span></div><div class=\\"at-button__text\\">按钮</div></div>"`;

exports[`AtButton Snap render AtButton -- props size(normal) 1`] = `"<div class=\\"at-button at-button--normal\\">undefined<div class=\\"at-button__text\\">按钮</div></div>"`;

exports[`AtButton Snap render AtButton -- props size(small) 1`] = `"<div class=\\"at-button at-button--small\\">undefined<div class=\\"at-button__text\\">按钮</div></div>"`;

exports[`AtButton Snap render AtButton -- props type(primary) 1`] = `"<div class=\\"at-button at-button--normal at-button--primary\\">undefined<div class=\\"at-button__text\\">按钮</div></div>"`;

exports[`AtButton Snap render AtButton -- props type(secondary) 1`] = `"<div class=\\"at-button at-button--normal at-button--secondary\\">undefined<div class=\\"at-button__text\\">按钮</div></div>"`;
